meshnets/utils/callbacks.py

```diff
@@ -51,6 +51,32 @@ def finalize(self, status: str) -> None:
 class GPUUsage(Callback):
     """Track and log GPU usage."""
 
+    def __init__(self, log_freq=50):
+        """
+        Args:
+            log_freq: The frequency, in batches, at which to log GPU usage.
+        """
+        self.log_freq = log_freq
+        self.gpu_usage_history = {}
+
+    def on_batch_end(self, trainer, pl_module):
+        if (trainer.global_step +
+                1) % self.log_freq == 0 or trainer.is_last_batch:
```

Comment:
Shall we log at the first batch as well?

Reply:
<img width="714" alt="Screenshot 2023-08-11 at 11 20 52" src="https://github.com/inductiva/meshnets/assets/83284015/6af92992-1675-4399-9e05-a59a413914b9">

I tried to follow lightning implementation in the class LoggerConnector that Trainer uses. This way the callback logs are aligned with the training step logs.

Reply:
Seems good!